core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-tables/contributor-admin-stats-table.component.ts

```diff
@@ -0,0 +1,244 @@
+      state('collapsed', style({ transform: 'rotate(0)' })),
+      transition('expanded => collapsed', animate('200ms ease-out')),
+      transition('collapsed => expanded', animate('200ms ease-in')),
+    ]),
+    trigger('mobileChevronExpand', [
+      state('expanded', style({ transform: 'rotate(-90deg)' })),
+      state('collapsed', style({ transform: 'rotate(90deg)' })),
+      transition('expanded => collapsed', animate('200ms ease-out')),
+      transition('collapsed => expanded', animate('200ms ease-in')),
+    ]),
+  ],
+})
+export class ContributorAdminStatsTable implements OnInit {
+  @Input() activeTab: string = 'Translation Submitter';
+
+  columnsToDisplay = [
+    'chevron',
+    'contributorName',
+    'recentPerformance',
+    'overallAccuracy',
+    'submittedTranslationsCount',
+    'lastContributedInDays',
+    'role'
+  ];
+
+  dataSource: TranslationSubmitterStats[] |
+    TranslationReviewerStats[] |
+    QuestionSubmitterStats[] |
+    QuestionReviewerStats[] = [];
+
+  nextOffset: number = 0;
+  more: boolean = false;
+
+  expandedElement: TranslationSubmitterStats[] |
+    TranslationReviewerStats[] |
+    QuestionSubmitterStats[] |
+    QuestionReviewerStats[] | null | [] = null;
+
+  TAB_NAME_TRANSLATION_SUBMITTER: string = 'Translation Submitter';
+  TAB_NAME_TRANSLATION_REVIEWER: string = 'Translation Reviewer';
+  TAB_NAME_QUESTION_SUBMITTER: string = 'Question Submitter';
+  TAB_NAME_QUESTION_REVIEWER: string = 'Question Reviewer';
+  TAB_NAME_LANGUAGE_COORDINATOR: string = 'Language Coordinator';
+  TAB_NAME_QUESTION_COORDINATOR: string = 'Question Coordinator';
+  loadingMessage!: string;
+
+  constructor(
+    private windowRef: WindowRef,
+    private ContributorDashboardAdminStatsBackendApiService:
+      ContributorDashboardAdminStatsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    this.loadingMessage = 'Loading';
+    this.updateColumnsToDisplay();
+  }
+
+  checkMobileView(): boolean {
+    return (this.windowRef.nativeWindow.innerWidth < 800);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.activeTab) {
+      this.loadingMessage = 'Loading';
+      this.updateColumnsToDisplay();
+    }
+  }
+
+  updateColumnsToDisplay(): void {
+    if (this.activeTab === this.TAB_NAME_TRANSLATION_SUBMITTER) {
+      this.columnsToDisplay = [
+        'chevron',
+        'contributorName',
+        'recentPerformance',
+        'overallAccuracy',
+        'submittedTranslationsCount',
+        'lastContributedInDays',
+        'role'
+      ];
+      if (this.checkMobileView()) {
+        this.columnsToDisplay = [
+          'contributorName',
+          'recentPerformance',
+          'overallAccuracy',
+          'submittedTranslationsCount',
+          'lastContributedInDays',
+          'chevron'
+        ];
+      }
+      this.ContributorDashboardAdminStatsBackendApiService
+        .fetchContributorAdminStats(
+          ContributorAdminDashboardFilter.createDefault(),
+          20,
```

Comment:
Why 20? Can it be a const?

Reply:
It is defaulted for now, this will change according to user preference.All the pagination, sorting and filtering from frontend is hardcoded for now.